src/DataProvider/Pagination.php

```diff
@@ -196,6 +196,21 @@ public function isPartialEnabled(string $resourceClass = null, string $operation
         return $this->getEnabled($context, $resourceClass, $operationName, true);
     }
 
+    public function getItemsPerPageOptions(): array
+    {
+        return array_intersect_key($this->options, array_flip([
+            'client_items_per_page',
+            'items_per_page_parameter_name',
+        ]));
+    }
+
+    public function getGraphQlPaginationType(string $resourceClass = null, string $operationName = null): string
+    {
+        $resourceMetadata = $this->resourceMetadataFactory->create($resourceClass);
+
+        return (string) $resourceMetadata->getGraphqlAttribute($operationName, 'paginationType', 'cursor', true);
+    }
```

Comment:
An if `$resourceClass` or `$operationName` is `null`? Maybe they should not be nullable?